homeassistant/components/zwave/node_entity.py

```diff
@@ -192,6 +196,42 @@ def node_changed(self):
 
         self.maybe_schedule_update()
 
+    async def node_renamed(self, update_ids=False):
+        """Rename the node and update any IDs."""
+        self._name = node_name(self.node)
+        # Set the name in the devices. If they're customised
+        # the customisation will not be stored as name and will stick.
+        dev_reg = await get_dev_reg(self.hass)
+        device = dev_reg.async_get_device(
+            identifiers={(DOMAIN, self.node_id), },
+            connections=set())
+        dev_reg.async_update_device(device.id, name=self._name)
+        # update sub-devices too
+        for i in count(2):
```

Comment:
Is there a max of 2 subdevices ?

Reply:
Starts with 2, indefinite quantity. 1 doesn't have an ID suffix.

Reply:
oh right. I was thinking of `range`